streams/src/main/java/org/apache/kafka/streams/processor/Processor.java

```diff
@@ -57,5 +57,5 @@
      * <p>
      * Note: Do not close any streams managed resources, like {@link StateStore}s here, as they are managed by the library.
      */
-    void close();
+    default void close() {}
```

Comment:
Needs KIP: I added defaults to the old Processor as well as the new TypedProcessor. I don't think this impacts compatibility.